From 64759d0ea3d36292ddd8fd57417583c236f38ffd Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Mon, 20 Jul 2015 14:10:25 -0700
Subject: [PATCH 05/23] gstomxvideodec: associate GstIonBufFdMeta data with OMX
 output buffer

---
 omx/gstomxvideodec.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index f04e093..71f8941 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -481,6 +481,8 @@ GstIonBufFdMeta * gst_buffer_add_ionBufFd_meta (GstBuffer *buffer, unsigned int
   meta->offset = offset;
   meta->size = size;
 
+  g_print ("Adding ionBufFd metadata fd %u, offset %u, size %u\n", fd, offset, size);
+
   return meta;
 }
 #endif
@@ -498,6 +500,7 @@ gst_omx_video_dec_fill_buffer (GstOMXVideoDec * self,
 
 #if (defined (USE_OMX_TARGET_VENUS))
   GstOmxPlatformPrivateInfo *ionBufInfo = NULL;
+  GstIonBufFdMeta *meta;
 #endif
 
   if (vinfo->width != port_def->format.video.nFrameWidth ||
@@ -519,6 +522,17 @@ gst_omx_video_dec_fill_buffer (GstOMXVideoDec * self,
   }
 
   GST_DEBUG_OBJECT (self, "Found ionBufInfo in gst_omx_video_dec_fill_buffer(): size %u, offset %u and fd %u\n", ionBufInfo->ion_buf_size, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_fd);
+
+  meta = gst_buffer_add_ionBufFd_meta(outbuf, ionBufInfo->ion_buf_fd, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_size);
+
+  if (!meta) {
+    GST_ERROR_OBJECT (self, "Addition of ionBufInfo metadata to decoder output buffer failed.\n");
+    ret = FALSE;
+    goto done;
+  }
+
+  ret = TRUE;
+  goto done;
 #endif
 
   /* Same strides and everything */
-- 
1.9.1

