From cded0c2053806c7c5a6bcaa75455811370a8540d Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Wed, 29 Jul 2015 10:00:57 -0700
Subject: [PATCH 08/23] gstomxvideodec: remove release_buffer from dec_loop

---
 omx/gstomxvideodec.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index 755fbaf..0bb51e7 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -1548,11 +1548,16 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
 
   GST_DEBUG_OBJECT (self, "Finished frame: %s", gst_flow_get_name (flow_ret));
 
+#if ((port->port_def.eDir == OMX_DirInput) ||  (!defined (USE_OMX_TARGET_VENUS)))
+    /* wait for gstwaylandsink pad event */
   if (buf) {
     err = gst_omx_port_release_buffer (port, buf);
     if (err != OMX_ErrorNone)
       goto release_error;
   }
+#else
+  GST_DEBUG_OBJECT(self, "Not releasing output buffer in gst_omx_video_dec_loop");
+#endif
 
   self->downstream_flow_ret = flow_ret;
 
@@ -2422,10 +2427,17 @@ gst_omx_video_dec_handle_frame (GstVideoDecoder * decoder,
       buf->omx_buf->nTickCount = 0;
 
       self->started = TRUE;
+#if ((port->port_def.eDir == OMX_DirInput) ||  (!defined (USE_OMX_TARGET_VENUS)))
+    /* wait for gstwaylandsink pad event */
       err = gst_omx_port_release_buffer (port, buf);
       gst_buffer_replace (&self->codec_data, NULL);
       if (err != OMX_ErrorNone)
         goto release_error;
+#else
+      GST_DEBUG_OBJECT(self, "Not releasing output buffer in gst_omx_video_dec_loop");
++#endif
+
+#endif
       /* Acquire new buffer for the actual frame */
       continue;
     }
-- 
1.9.1

