From ebaa77b5194e74bfcdc74a219b43759029336db6 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Thu, 30 Jul 2015 10:25:38 -0700
Subject: [PATCH 09/23] gstomxvideodec: include GstOMXBuffer pointer in
 GstIonBufFdMeta

---
 omx/gstomxvideodec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index 0bb51e7..c1b9b87 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -463,7 +463,7 @@ gst_omx_video_dec_fill_buffer (GstOMXVideoDec * self,
 
   GST_DEBUG_OBJECT (self, "Found ionBufInfo in gst_omx_video_dec_fill_buffer(): size %u, offset %u and fd %u\n", ionBufInfo->ion_buf_size, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_fd);
 
-  meta = gst_buffer_add_ionBufFd_meta(outbuf, ionBufInfo->ion_buf_fd, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_size);
+  meta = gst_buffer_add_ionBufFd_meta(outbuf, ionBufInfo->ion_buf_fd, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_size, inbuf);
 
   if (!meta) {
     GST_ERROR_OBJECT (self, "Addition of ionBufInfo metadata to decoder output buffer failed.\n");
-- 
1.9.1

