From a6de0d9260adab6a2797da29814b4318cf86b082 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Sat, 1 Aug 2015 00:48:38 -0700
Subject: [PATCH 12/23] gstomx: recycle OMXBuffer on wayland release event

---
 omx/gstomx.c         | 164 ++++++++++++++++++++++++++++++++-------------------
 omx/gstomxvideodec.c |  90 +++++++++++++++++-----------
 2 files changed, 161 insertions(+), 93 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 5dced58..0ab7ad7 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -50,6 +50,7 @@
 #if defined (USE_OMX_TARGET_VENUS)
 #include "gstomx_meta.h"
 #include <gst/gstevent.h>
+#include <gst/gststructure.h>
 #include <gst/ionbuf/gstionbuf_meta.h>
 #endif
 
@@ -379,7 +380,7 @@ gst_omx_component_handle_messages (GstOMXComponent * comp)
         } else {
           /* Output buffer contains output now or
            * the port was flushed */
-          GST_LOG_OBJECT (port->comp->parent,
+          GST_DEBUG_OBJECT (port->comp->parent,
               "%s port %u filled buffer %p (%p)", port->comp->name, port->index,
               buf, buf->omx_buf->pBuffer);
 
@@ -590,44 +591,78 @@ EventHandler (OMX_HANDLETYPE hComponent, OMX_PTR pAppData, OMX_EVENTTYPE eEvent,
 
 #if defined (USE_OMX_TARGET_VENUS)
 gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEvent * event) {
-  GstOMXBuffer * ionbufp = NULL;
-   GstOMXPort * outport = NULL;
-   const GstStructure * s;
-   gboolean ret = FALSE;
-
-   switch (GST_EVENT_TYPE (event)) {
-   case GST_EVENT_CUSTOM_UPSTREAM:
-     s = gst_event_get_structure (event);
-     if (!GST_IS_STRUCTURE(s)) {
-       GST_ERROR_OBJECT(self, "Event has no associated metadata.");
-       break;
-     }
-     if (gst_structure_has_name((GstStructure *) s, "IonBufRelease")) {
-       ionbufp = (GstOMXBuffer *) ((const GstIonBufFdMeta * ) s)->ionbufp;
-       outport = ((GstOMXVideoDec *) self)->dec_out_port;
-       if (!ionbufp) {
-	 GST_ERROR_OBJECT(self, "No readable GstIonBufMeta data.");
-	 break;
-       }
-       if (!outport)  {
-	 GST_ERROR_OBJECT(self, "Unable to look up decoder port.");
-	 break;
-       }
-       ret = gst_omx_port_release_buffer(outport, ionbufp);
-       if (ret) {
-	 GST_ERROR_OBJECT(self, "Unable to release OMX buffer.");
-	 break;
-       }
-     }
-     GST_DEBUG_OBJECT(self, "CUSTOM_UPSTREAM event has name %s.", gst_structure_get_name((GstStructure *) s));
-     gst_event_unref(event);
-     break;
-   default:
-     ret = gst_pad_event_default (pad,  self, event);
-     break;
-   }
-   return ret;
- }
+  GstOMXBuffer * ionbufp = g_malloc(sizeof *ionbufp);
+  GstOMXPort * outport = g_malloc(sizeof *outport);
+  GstOMXComponent *comp = g_malloc(sizeof *comp);
+  const GstStructure * s;
+  gboolean ret = FALSE;
+  OMX_ERRORTYPE err =  OMX_ErrorNone;
+  const GValue *ionbufval;
+
+  switch (GST_EVENT_TYPE (event)) {
+  case GST_EVENT_CUSTOM_UPSTREAM:
+    s = gst_event_get_structure (event);
+    if (!GST_IS_STRUCTURE(s)) {
+      GST_ERROR_OBJECT(self, "Event has no associated metadata.");
+      break;
+    }
+    if (gst_structure_has_name(s, "IonBufRelease")) {
+      ionbufval = gst_structure_get_value(s, "ion_buf_pointer");
+      if (!(G_IS_VALUE (ionbufval)))
+	{
+	  GST_ERROR("No GValue in event structure passed by waylandsink.");
+	  return FALSE;
+	}
+      ionbufp = (GstOMXBuffer *) g_value_get_pointer(ionbufval);
+
+      outport = ((GstOMXVideoDec *) self)->dec_out_port;
+      g_print("Processing GST_EVENT_CUSTOM_UPSTREAM with ionbufp %p and outport %p.\n", ionbufp, outport);
+      if (!ionbufp) {
+	GST_ERROR_OBJECT(self, "No readable GstIonBufMeta data.");
+	break;
+      }
+      if (!ionbufp->omx_buf) {
+	GST_ERROR_OBJECT(self, "No omx_buf in (GstOMXBuffer *) %p.", ionbufp);
+	break;
+      }
+      if (!outport)  {
+	GST_ERROR_OBJECT(self, "Unable to look up decoder port.");
+	break;
+      }
+      comp = outport->comp;
+      if (!comp)  {
+	GST_ERROR_OBJECT(self, "Unable to look up decoder component.");
+	break;
+      }
+
+      /* FIXME: What if the settings cookies don't match? */
+      g_assert (buf == buf->omx_buf->pAppPrivate);
+
+      /* Reset all flags, some implementations don't
+       * reset them themselves and the flags are not
+       * valid anymore after the buffer was consumed
+       */
+      ionbufp->omx_buf->nFlags = 0;
+
+      /* Reset offset and filled length */
+      ionbufp->omx_buf->nOffset = 0;
+      ionbufp->omx_buf->nFilledLen = 0;
+
+      ionbufp->used = TRUE;
+      g_print("omx_port_out_buf_release_event: calling OMX_FillThisBuffer with GstOMXBuffer * %p and OMX_BUFFERHEADERTYPE %p\n", ionbufp, ionbufp->omx_buf);
+      err = OMX_FillThisBuffer (comp->handle, ionbufp->omx_buf);
+      g_print("Recycled output buffer %p to %s port %u: %s \n", ionbufp->omx_buf,  comp->name, outport->index, gst_omx_error_to_string (err));
+      break;
+    }
+    GST_DEBUG_OBJECT(self, "CUSTOM_UPSTREAM event has name %s.", gst_structure_get_name((GstStructure *) s));
+    gst_event_unref(event);
+    break;
+  default:
+    ret = gst_pad_event_default (pad,  self, event);
+    break;
+  }
+  return ret;
+}
 
 static void findMeta (OMX_BUFFERHEADERTYPE *bufhead) {
 
@@ -1455,18 +1490,6 @@ gst_omx_port_release_buffer (GstOMXPort * port, GstOMXBuffer * buf)
 
   gst_omx_component_handle_messages (comp);
 
-  if (port->port_def.eDir == OMX_DirOutput) {
-    /* Reset all flags, some implementations don't
-     * reset them themselves and the flags are not
-     * valid anymore after the buffer was consumed
-     */
-    buf->omx_buf->nFlags = 0;
-
-    /* Reset offset and filled length */
-    buf->omx_buf->nOffset = 0;
-    buf->omx_buf->nFilledLen = 0;
-  }
-
   if ((err = comp->last_error) != OMX_ErrorNone) {
     GST_ERROR_OBJECT (comp->parent, "Component %s is in error state: %s "
         "(0x%08x)", comp->name, gst_omx_error_to_string (err), err);
@@ -1488,16 +1511,39 @@ gst_omx_port_release_buffer (GstOMXPort * port, GstOMXBuffer * buf)
 
   /* FIXME: What if the settings cookies don't match? */
 
-  buf->used = TRUE;
-
   if (port->port_def.eDir == OMX_DirInput) {
     err = OMX_EmptyThisBuffer (comp->handle, buf->omx_buf);
-  } else {
+    GST_DEBUG_OBJECT (comp->parent, "Released input buffer %p to %s port %u: %s "
+		      "(0x%08x)", buf, comp->name, port->index, gst_omx_error_to_string (err),
+		      err);
+  }
+
+  /* for Venus output buffer, wait for gstwaylandsink pad event to reuse buffer */
+#if defined (USE_OMX_TARGET_VENUS)
+  if (port->port_def.eDir == OMX_DirOutput)
+    g_print("Waiting for wayland: buf %p, buf->omx_buf %p, buf->omx_buf->pBuffer %p, nAllocLen %u, nFilledLen %u, nOffset %u",
+		     buf, buf->omx_buf, buf->omx_buf->pBuffer, (unsigned) buf->omx_buf->nAllocLen, (unsigned) buf->omx_buf->nFilledLen,
+		     (unsigned) buf->omx_buf->nOffset);
+#else
+  buf->used = TRUE;
+
+  if (port->port_def.eDir == OMX_DirOutput) {
+    /* Reset all flags, some implementations don't
+     * reset them themselves and the flags are not
+     * valid anymore after the buffer was consumed
+     */
+    buf->omx_buf->nFlags = 0;
+
+    /* Reset offset and filled length */
+    buf->omx_buf->nOffset = 0;
+    buf->omx_buf->nFilledLen = 0;
+
     err = OMX_FillThisBuffer (comp->handle, buf->omx_buf);
+    GST_DEBUG_OBJECT (comp->parent, "Released output buffer %p to %s port %u: %s "
+		      "(0x%08x)", buf, comp->name, port->index, gst_omx_error_to_string (err),
+		      err);
   }
-  GST_DEBUG_OBJECT (comp->parent, "Released buffer %p to %s port %u: %s "
-      "(0x%08x)", buf, comp->name, port->index, gst_omx_error_to_string (err),
-      err);
+#endif
 
 done:
   gst_omx_component_handle_messages (comp);
@@ -2712,9 +2758,7 @@ plugin_init (GstPlugin * plugin)
   static const gchar *env_config_name[] = { "GST_OMX_CONFIG_DIR", NULL };
 
   GST_DEBUG_CATEGORY_INIT (gstomx_debug, "omx", 0, "gst-omx");
-#define DEBUG_INIT \
-  GST_DEBUG_CATEGORY_INIT (gst_omx_video_debug_category, "omxvideo", 0,
-      "gst-omx-video");
+  GST_DEBUG_CATEGORY_INIT (gst_omx_video_debug_category, "omxvideo", 0, "gst-omx-video");
 
   /* Read configuration file gstomx.conf from the preferred
    * configuration directories */
diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index 73a7821..60ca7da 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -461,7 +461,7 @@ gst_omx_video_dec_fill_buffer (GstOMXVideoDec * self,
     goto done;
   }
 
-  GST_DEBUG_OBJECT (self, "Found ionBufInfo in gst_omx_video_dec_fill_buffer(): size %u, offset %u and fd %u\n", ionBufInfo->ion_buf_size, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_fd);
+  g_print ("Found ionBufInfo in gst_omx_video_dec_fill_buffer(): size %u, offset %u and fd %u\n", ionBufInfo->ion_buf_size, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_fd);
 
   meta = gst_buffer_add_ionBufFd_meta(outbuf, ionBufInfo->ion_buf_fd, ionBufInfo->ion_buf_offset, ionBufInfo->ion_buf_size, inbuf);
 
@@ -1551,23 +1551,11 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
 
   GST_DEBUG_OBJECT (self, "Finished frame: %s", gst_flow_get_name (flow_ret));
 
-  g_assert (port->port_def.eDir == OMX_DirOutput);
-
-  /* for Venus output buffer, wait for gstwaylandsink pad event */
-#if (!defined (USE_OMX_TARGET_VENUS))
   if (buf) {
     err = gst_omx_port_release_buffer (port, buf);
     if (err != OMX_ErrorNone)
       goto release_error;
   }
-#else
-  g_mutex_lock (&(port->comp->lock));
-  gst_omx_component_handle_messages (port->comp);
-  buf->used = TRUE;
-  g_mutex_unlock (&(port->comp->lock));
-  GST_DEBUG_OBJECT(self, "Not releasing output buffer %p, nAllocLen %u, nFilledLen %u, nOffset %u",
-		   buf->omx_buf, buf->omx_buf->nAllocLen, buf->omx_buf->nFilledLen, buf->omx_buf->nOffset);
-#endif
 
   self->downstream_flow_ret = flow_ret;
 
@@ -1841,6 +1829,8 @@ gst_omx_video_dec_negotiate (GstOMXVideoDec * self)
   GST_OMX_INIT_STRUCT (&param);
   param.eColorFormat = OMX_COLOR_FormatYUV420SemiPlanar;
   param.nPortIndex = self->dec_out_port->index;
+  param.eCompressionFormat = OMX_VIDEO_CodingUnused;
+  param.xFramerate = 25;
 #else
   intersection =
       gst_pad_peer_query_caps (GST_VIDEO_DECODER_SRC_PAD (self), templ_caps);
@@ -1916,7 +1906,14 @@ gst_omx_video_dec_negotiate (GstOMXVideoDec * self)
         gst_omx_error_to_string (err), err);
   }
   else {
-    GST_DEBUG_OBJECT (self, "OMX_IndexParamVideoPortFormat %u : port %u, coding %#x, colorformat %s, framerate %#x \n",
+  err = gst_omx_component_get_parameter (self->dec, OMX_IndexParamVideoPortFormat,
+      &param);
+  if (err != OMX_ErrorNone) {
+    GST_ERROR_OBJECT (self, "failed to add port %u: %s (0x%08x)",
+        (unsigned) param.nPortIndex, gst_omx_error_to_string (err), err);
+    return err;
+  }
+    GST_DEBUG_OBJECT (self, "Actual OMX_IndexParamVideoPortFormat %u : port %u, coding %#x, colorformat %s, framerate %#x \n",
 	   (unsigned int) param.nIndex, (unsigned int) param.nPortIndex, param.eCompressionFormat, decode_color(param.eColorFormat), (unsigned int) param.xFramerate );
   }
 
@@ -2124,13 +2121,8 @@ gst_omx_video_dec_set_format (GstVideoDecoder * decoder,
       return FALSE;
   } else {
 #if defined(USE_OMX_TARGET_VENUS)
-    /*
-      Decoder will render into dmabufs and does not require output buffer
-      allocation.  Format negotiation is not needed in the case of
-      Venus.  Instead, set new SRC pad event function for the purpose of
-      Wayland buffer-release synchronization.
-    */
-  gst_pad_set_event_function (srcpad,  (GstPadEventFunction) GST_DEBUG_FUNCPTR (omx_port_out_buf_release_event));
+    /* set new SRC pad event function for the purpose of Wayland buffer-release synchronization */
+    gst_pad_set_event_function (srcpad,  (GstPadEventFunction) GST_DEBUG_FUNCPTR (omx_port_out_buf_release_event));
 #endif
     if (!gst_omx_video_dec_negotiate (self))
       GST_LOG_OBJECT (self, "Negotiation failed, will get output format later");
@@ -2292,6 +2284,7 @@ gst_omx_video_dec_handle_frame (GstVideoDecoder * decoder,
   guint offset = 0, size;
   GstClockTime timestamp, duration;
   OMX_ERRORTYPE err;
+  gsize copied = 0;
 
   self = GST_OMX_VIDEO_DEC (decoder);
   klass = GST_OMX_VIDEO_DEC_GET_CLASS (self);
@@ -2305,6 +2298,34 @@ gst_omx_video_dec_handle_frame (GstVideoDecoder * decoder,
   }
 
   if (!self->started) {
+#if defined (USE_OMX_TARGET_VENUS)
+    /*
+      Initialize output port with first data and call OMX_FillThisBuffer once.   After the first call,
+      FillThisBuffer will only be triggered by release events originating with Wayland.
+    */
+    err = gst_omx_port_populate (self->dec_out_port);
+    if (err != OMX_ErrorNone) {
+      GST_WARNING_OBJECT (self, "Failed to populate output port: %s (0x%08x)",
+			  gst_omx_error_to_string (err), err);
+    }
+
+      err = gst_omx_port_set_enabled (self->dec_out_port, TRUE);
+      if (err != OMX_ErrorNone)
+	GST_INFO_OBJECT (self,
+			 "Failed to enable port: %s (0x%08x)",
+			 gst_omx_error_to_string (err), err);
+
+      err = gst_omx_port_wait_enabled (self->dec_out_port, 1 * GST_SECOND);
+      if (err != OMX_ErrorNone) {
+	GST_ERROR_OBJECT(self, "Output port enablement failure.");
+      return GST_FLOW_ERROR;
+    }
+      /*    err = gst_omx_port_mark_reconfigured (port);
+    if (err !=  OMX_ErrorNone) {
+      GST_ERROR_OBJECT(self, "Output port reconfiguration failure.");
+      return GST_FLOW_ERROR;
+      } */
+#endif
     if (!GST_VIDEO_CODEC_FRAME_IS_SYNC_POINT (frame)) {
       gst_video_decoder_drop_frame (GST_VIDEO_DECODER (self), frame);
       return GST_FLOW_OK;
@@ -2447,27 +2468,21 @@ gst_omx_video_dec_handle_frame (GstVideoDecoder * decoder,
 
       self->started = TRUE;
 
-      g_assert (port->port_def.eDir == OMX_DirOutput);
+  GST_DEBUG_OBJECT(self, "Releasing port input buffer %p, nAllocLen %u, nFilledLen %u, nOffset %u",
+		   buf->omx_buf, (unsigned) buf->omx_buf->nAllocLen, (unsigned) buf->omx_buf->nFilledLen, (unsigned) buf->omx_buf->nOffset);
 
-      /* for Venus outbuf wait for gstwaylandsink pad event */
-#if (!defined (USE_OMX_TARGET_VENUS))
       err = gst_omx_port_release_buffer (port, buf);
       gst_buffer_replace (&self->codec_data, NULL);
       if (err != OMX_ErrorNone)
         goto release_error;
-#else
-      g_mutex_lock (&(port->comp->lock));
-      gst_omx_component_handle_messages (port->comp);
-      buf->used = TRUE;
-      g_mutex_unlock (&(port->comp->lock));
-      GST_DEBUG_OBJECT(self, "Not releasing output buffer %p, nAllocLen %u, nFilledLen %u, nOffset %u",
-		   buf->omx_buf, buf->omx_buf->nAllocLen, buf->omx_buf->nFilledLen, buf->omx_buf->nOffset);
-#endif
 
       /* Acquire new buffer for the actual frame */
       continue;
     }
 
+      GST_DEBUG_OBJECT(self, "Buffer %p, nAllocLen %u, nFilledLen %u, nOffset %u",
+		       buf->omx_buf, (unsigned) buf->omx_buf->nAllocLen, (unsigned) buf->omx_buf->nFilledLen, (unsigned) buf->omx_buf->nOffset);
+
     /* Now handle the frame */
     GST_DEBUG_OBJECT (self, "Passing frame offset %d to the component", offset);
 
@@ -2475,10 +2490,19 @@ gst_omx_video_dec_handle_frame (GstVideoDecoder * decoder,
      * by the port */
     buf->omx_buf->nFilledLen =
         MIN (size - offset, buf->omx_buf->nAllocLen - buf->omx_buf->nOffset);
-    gst_buffer_extract (frame->input_buffer, offset,
+    /* copy nFilledLen bytes from  (frame->input_buffer + offset) to
+       (buf->omx_buf->pBuffer + buf->omx_buf->nOffset)
+    */
+    copied = gst_buffer_extract (frame->input_buffer, offset,
         buf->omx_buf->pBuffer + buf->omx_buf->nOffset,
         buf->omx_buf->nFilledLen);
 
+    if (copied < buf->omx_buf->nFilledLen) {
+      GST_ERROR_OBJECT(self, "Wanted %u bytes from frame, got %u.", (unsigned) buf->omx_buf->nFilledLen, copied);
+      return GST_FLOW_ERROR;
+    }
+    GST_DEBUG_OBJECT(self, "Copied %u bytes from input frame to port's buffer.",  copied);
+
     if (timestamp != GST_CLOCK_TIME_NONE) {
       buf->omx_buf->nTimeStamp =
           gst_util_uint64_scale (timestamp, OMX_TICKS_PER_SECOND, GST_SECOND);
-- 
1.9.1

