From f1bc8a45531984ae76c603d05bc8f1a9cc3c7630 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Wed, 5 Aug 2015 00:18:45 -0700
Subject: [PATCH 13/23] gstomxvideodec: no buffer deallocation for deactivated
 port

---
 omx/gstomxvideodec.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index 60ca7da..9dbe733 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -307,7 +307,13 @@ gst_omx_video_dec_shutdown (GstOMXVideoDec * self)
     }
     gst_omx_component_set_state (self->dec, OMX_StateLoaded);
     gst_omx_port_deallocate_buffers (self->dec_in_port);
-    gst_omx_video_dec_deallocate_output_buffers (self);
+    if (self->dec_out_port)
+      gst_omx_video_dec_deallocate_output_buffers (self);
+    else {
+      g_print("Output port already unpopulated; skip releasing output buffers.");
+      return TRUE;
+    }
+
     if (state > OMX_StateLoaded)
       gst_omx_component_get_state (self->dec, 5 * GST_SECOND);
   }
-- 
1.9.1

