From 01c6e94c383f6a4b892ddb1a88675a58700f2bb1 Mon Sep 17 00:00:00 2001
From: Mathias Fiedler <mathias_fiedler@mentor.com>
Date: Mon, 17 Aug 2015 15:08:06 +0200
Subject: [PATCH 22/23] hack: wait 30ms before reusing unreleased buffers

---
 omx/gstomx.c | 29 ++++++++++++++---------------
 1 file changed, 14 insertions(+), 15 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 1fc6366..57522dd 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -1498,30 +1498,29 @@ retry:
         comp->name, port->index);
 
     /* HACK: */
-    if ((_buf = g_queue_pop_head (&port->pushed_buffers)) != NULL)
+    if (! gst_omx_component_wait_message (comp, GST_MSECOND * 30))
     {
-      _buf->used = TRUE;
+      while ((_buf = g_queue_pop_head (&port->pushed_buffers)) != NULL) {
+        _buf->used = TRUE;
 
-      /* Reset all flags, some implementations don't
-       * reset them themselves and the flags are not
-       * valid anymore after the buffer was consumed
-       */
-      _buf->omx_buf->nFlags = 0;
+        /* Reset all flags, some implementations don't
+         * reset them themselves and the flags are not
+         * valid anymore after the buffer was consumed
+         */
+        _buf->omx_buf->nFlags = 0;
 
-      /* Reset offset and filled length */
-      _buf->omx_buf->nOffset = 0;
-      _buf->omx_buf->nFilledLen = 0;
+        /* Reset offset and filled length */
+        _buf->omx_buf->nOffset = 0;
+        _buf->omx_buf->nFilledLen = 0;
 
-      GST_ERROR_OBJECT(comp->parent, "reusing unreleased GstOMXBuffer %p ", _buf);
+        GST_ERROR_OBJECT(comp->parent, "reusing unreleased GstOMXBuffer %p ", _buf);
 
-      OMX_FillThisBuffer (comp->handle, _buf->omx_buf);
-      // gst_omx_component_handle_messages (comp);
+        OMX_FillThisBuffer (comp->handle, _buf->omx_buf);
+      }
 
       _buf = NULL;
     }
 
-
-    gst_omx_component_wait_message (comp, GST_CLOCK_TIME_NONE);
     gst_omx_component_handle_messages (comp);
 
     /* And now check everything again and maybe get a buffer */
-- 
1.9.1

