From 6e16e254fd3d2ea1276ec8687ec2a8bd2a9de910 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Mon, 24 Aug 2015 16:41:53 -0700
Subject: [PATCH 23/23] free wl_omx_buffer_release event in omx handler

---
 omx/gstomx.c | 36 ++++++++++++++++++++++--------------
 1 file changed, 22 insertions(+), 14 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 57522dd..8e70d67 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -618,9 +618,9 @@ EventHandler (OMX_HANDLETYPE hComponent, OMX_PTR pAppData, OMX_EVENTTYPE eEvent,
 #if defined (USE_OMX_TARGET_VENUS)
 /* NOTE: Uses comp->lock and comp->messages_lock */
 gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEvent * event) {
-  GstOMXBuffer * ionbufp = g_malloc(sizeof *ionbufp);
-  GstOMXPort * outport = g_malloc(sizeof *outport);
-  GstOMXComponent *comp = g_malloc(sizeof *comp);
+  GstOMXBuffer * ionbufp = NULL;
+  GstOMXPort * outport = NULL;
+  GstOMXComponent *comp = NULL;
   const GstStructure * s;
   gboolean ret = FALSE;
   OMX_ERRORTYPE err =  OMX_ErrorNone;
@@ -631,7 +631,8 @@ gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEven
     s = gst_event_get_structure (event);
     if (!GST_IS_STRUCTURE(s)) {
       GST_ERROR_OBJECT(self, "Event has no associated metadata.");
-      break;
+      gst_event_unref(event);
+      return ret;
     }
     if (gst_structure_has_name(s, "IonBufRelease")) {
 
@@ -639,28 +640,29 @@ gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEven
       if (!(G_IS_VALUE (ionbufval)))
 	{
 	  GST_ERROR("No GValue in event structure passed by waylandsink.");
-	  ret = FALSE;
+	  goto done;
 	}
       ionbufp = (GstOMXBuffer *) g_value_get_pointer(ionbufval);
+      g_value_unset(ionbufval);
 
       outport = ((GstOMXVideoDec *) self)->dec_out_port;
       GST_DEBUG_OBJECT(self, "Processing GST_EVENT_CUSTOM_UPSTREAM with ionbufp %p and outport %p.", ionbufp, outport);
       if (!ionbufp) {
 	GST_ERROR_OBJECT(self, "No readable GstIonBufMeta data.");
-	break;
+	goto done;
       }
       if (!ionbufp->omx_buf) {
 	GST_ERROR_OBJECT(self, "No omx_buf in (GstOMXBuffer *) %p.", ionbufp);
-	break;
+	goto done;
       }
       if (!outport)  {
 	GST_ERROR_OBJECT(self, "Unable to look up decoder port.");
-	break;
+	goto done;
       }
       comp = outport->comp;
       if (!comp)  {
 	GST_ERROR_OBJECT(self, "Unable to look up decoder component.");
-	break;
+	goto done;
       }
 
       /* NOTE: Uses comp->lock and comp->messages_lock in imitation of gst_omx_port_release_buffer */
@@ -671,7 +673,8 @@ gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEven
 			  "(0x%08x)", comp->name, gst_omx_error_to_string (err), err);
 	g_queue_push_tail (&outport->pending_buffers, ionbufp);
 	gst_omx_component_send_message (comp, NULL);
-	break;
+	g_mutex_unlock (&comp->lock);
+	goto done;
       }
 
       if (outport->flushing || outport->disabled_pending || !outport->port_def.bEnabled) {
@@ -680,7 +683,8 @@ gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEven
 			  comp->name, outport->index);
 	g_queue_push_tail (&outport->pending_buffers, ionbufp);
 	gst_omx_component_send_message (comp, NULL);
-	break;
+	g_mutex_unlock (&comp->lock);
+	goto done;
       }
 
       /* FIXME: What if the settings cookies don't match? */
@@ -709,16 +713,20 @@ gboolean omx_port_out_buf_release_event(GstPad * pad,  GstObject * self, GstEven
 
       GST_DEBUG_OBJECT(self, "Recycled output buffer %p to %s port %u: error %s", ionbufp->omx_buf,  comp->name, outport->index, gst_omx_error_to_string (err));
       ret = (err == OMX_ErrorNone);
-      break;
+      goto done;
     }
+  default:
     /* fall through to default case for non-buffer-release custom upstream event */
     GST_DEBUG_OBJECT(self, "Handling CUSTOM_UPSTREAM event with name %s.", gst_structure_get_name((GstStructure *) s));
-  default:
     ret = gst_pad_event_default (pad,  self, event);
-    break;
   }
 
   return ret;
+
+ done:
+  /* also frees the GstStructure contained inside the event */
+  gst_event_unref(event);
+  return ret;
 }
 
 static void findMeta (OMX_BUFFERHEADERTYPE *bufhead) {
-- 
1.9.1

