--- base-passwd/passwd.master.orig  2022-11-16 12:49:08.740208279 +0800
+++ base-passwd/passwd.master       2022-11-16 12:52:22.169421686 +0800
@@ -15,4 +15,5 @@
 list:*:38:38:Mailing List Manager:/var/list:/sbin/nologin
 irc:*:39:39:ircd:/var/run/ircd:/sbin/nologin
 gnats:*:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/sbin/nologin
+diag:*:53:53:diag:/nonexistent:/bin/sh
 nobody:*:65534:65534:nobody:/nonexistent:/sbin/nologin
