From e34aedc3906c9446cb69146a535814ecd5aa76d5 Mon Sep 17 00:00:00 2001
From: Kai Meng <kmeng@qti.qualcomm.com>
Date: Thu, 22 Jan 2026 11:28:12 +0800
Subject: [PATCH] libsemanage: Change sepolicy store-root path

Signed-off-by: Kai Meng <kmeng@qti.qualcomm.com>
---
 src/semanage.conf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/semanage.conf b/src/semanage.conf
index 708fa8cb..99ffaff9 100644
--- a/src/semanage.conf
+++ b/src/semanage.conf
@@ -37,6 +37,10 @@ module-store = direct
 # version is necessary.
 #policy-version = 19
 
+#change store-root = "/etc/selinux/policy-store"
+store-root = "/etc/selinux/policy-store"
+
+
 # By default, semanage will generate policies for the SELinux target.
 # To build policies for Xen, uncomment the following line.
 #target-platform = xen
-- 
2.34.1

