From d83cbbc638e1897965c3bc101b1c4cf33c17f693 Mon Sep 17 00:00:00 2001
From: Bharathraj Nagaraju <snbraj@codeaurora.org>
Date: Wed, 15 Feb 2017 13:09:45 -0800
Subject: [PATCH] common: oem-recovery : Add partition label path.

Signed-off-by: Bharathraj Nagaraju <snbraj@codeaurora.org>
---
 gpt-utils.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gpt-utils.c b/gpt-utils.c
index 3684817..c597cde 100644
--- a/gpt-utils.c
+++ b/gpt-utils.c
@@ -48,9 +48,9 @@
 /******************************************************************************
  * DEFINE SECTION
  ******************************************************************************/
-#define BLK_DEV_FILE    "/dev/block/mmcblk0"
-#define UFS_DEV_DIR     "/dev/block/sda"
-#define BOOT_DEV_DIR    "/dev/block/bootdevice/by-name"
+#define BLK_DEV_FILE    "/dev/mmcblk0"
+#define UFS_DEV_DIR     "/dev/sda"
+#define BOOT_DEV_DIR    "/dev/disk/by-partlabel"
 /* list the names of the backed-up partitions to be swapped */
 #define PTN_SWAP_LIST       "sbl1", "rpm", "tz", "aboot", "hyp", "lksecapp", "keymaster", "cmnlib", "cmnlib64", "pmic", "devcfg"
 /* extension used for the backup partitions - tzbak, abootbak, etc. */
-- 
2.9.0.137.gcf4c2cf

