From e708b1e7529bb4773ad07e2eeb4d025cd8183419 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Thu, 16 Jul 2015 18:04:57 +0200
Subject: [PATCH 06/26] gst-plugins-bad waylandsink window tile

---
 ext/wayland/wlwindow.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/wayland/wlwindow.c b/ext/wayland/wlwindow.c
index f827501..27944f6 100644
--- a/ext/wayland/wlwindow.c
+++ b/ext/wayland/wlwindow.c
@@ -125,6 +125,7 @@ GstWlWindow *
 gst_wl_window_new_toplevel (GstWlDisplay * display, GstVideoInfo * video_info)
 {
   GstWlWindow *window;
+  const char *title=NULL;
 
   window = gst_wl_window_new_internal (display,
       wl_compositor_create_surface (display->compositor));
@@ -136,6 +137,10 @@ gst_wl_window_new_toplevel (GstWlDisplay * display, GstVideoInfo * video_info)
   window->shell_surface = wl_shell_get_shell_surface (display->shell,
       window->surface);
 
+  title = getenv("GST_WINDOW_TITLE");
+  if( title )
+    wl_shell_surface_set_title(window->shell_surface, title);
+
   if (window->shell_surface) {
     wl_shell_surface_add_listener (window->shell_surface,
         &shell_surface_listener, window);
-- 
1.9.1

