From faa79ac82998ff9e0a1963aa4cbe2a3bbe056f5f Mon Sep 17 00:00:00 2001
From: Mathias Fiedler <mathias_fiedler@mentor.com>
Date: Fri, 31 Jul 2015 00:32:34 +0200
Subject: [PATCH 17/26] waylandsink: clean buffer_table only if it has been
 created

---
 ext/wayland/gstwaylandsink.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 9a9faa9..8fe3d43 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -450,7 +450,8 @@ gst_wayland_sink_change_state (GstElement * element, GstStateChange transition)
               (sink->pool));
         }
 
-        g_hash_table_remove_all(sink->buffer_table);
+        if (sink->buffer_table)
+          g_hash_table_remove_all(sink->buffer_table);
 
         g_clear_object (&sink->display);
         g_clear_object (&sink->pool);
-- 
1.9.1

