From 3e67596484e897235aaca26ccb4a99a6ceb1b24d Mon Sep 17 00:00:00 2001
From: Mathias Fiedler <mathias_fiedler@mentor.com>
Date: Fri, 31 Jul 2015 00:33:11 +0200
Subject: [PATCH 18/26] waylandsink: relax sink pad caps

---
 ext/wayland/gstwaylandsink.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 8fe3d43..b58227d 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -93,9 +93,7 @@ static GstStaticPadTemplate sink_template = GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
 #if (defined (USE_OMX_TARGET_VENUS))
-    GST_STATIC_CAPS ("video/x-raw, format = NV12,"
-      "width = 1920, "
-      "height = 1080, framerate = 25/1")
+    GST_STATIC_CAPS ("video/x-raw, format = NV12")
 #else
     GST_STATIC_CAPS (GST_VIDEO_CAPS_MAKE
         ("{ BGRx, BGRA, RGBx, xBGR, xRGB, RGBA, ABGR, ARGB, RGB, BGR, "
-- 
1.9.1

