From 27e34bd98df7adfdf3d0f02b7daeed2ca9549454 Mon Sep 17 00:00:00 2001
From: Mathias Fiedler <mathias_fiedler@mentor.com>
Date: Thu, 20 Aug 2015 16:54:34 +0200
Subject: [PATCH 25/26] waylandsink: wait for framecallback in preroll

---
 ext/wayland/gstwaylandsink.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 9b2685e..9b9bcba 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -735,8 +735,15 @@ config_failed:
 static GstFlowReturn
 gst_wayland_sink_preroll (GstBaseSink * bsink, GstBuffer * buffer)
 {
+  GstFlowReturn ret;
+  GstWaylandSink *sink = GST_WAYLAND_SINK (bsink);
   GST_DEBUG_OBJECT (bsink, "preroll buffer %p", buffer);
-  return gst_wayland_sink_render (bsink, buffer);
+  ret = gst_wayland_sink_render (bsink, buffer);
+
+  while (sink->redraw_pending)
+    wl_display_roundtrip(sink->display->display);
+
+  return ret;
 }
 
 static void
-- 
1.9.1

