From dd39f62fb8bec65353111150e453212bdf38ef19 Mon Sep 17 00:00:00 2001
From: Alison Chaiken <alison_chaiken@mentor.com>
Date: Mon, 24 Aug 2015 14:53:59 -0700
Subject: [PATCH 26/26] waylandsink: wl_omx_buffer_release cleanup

---
 ext/wayland/gstwaylandsink.c | 39 +++++++++++++++++----------------------
 1 file changed, 17 insertions(+), 22 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 9b9bcba..54af136 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -74,7 +74,7 @@ GST_DEBUG_CATEGORY (gstwayland_debug);
 #include <EGL/eglext.h>
 #include <drm/drm_fourcc.h>
 
-gboolean check_event_struct (const GstStructure *s, GstIonBufFdMeta *buf);
+gboolean check_event_struct (const GstStructure *s, gpointer buf);
 void wl_omx_buffer_release (void *data, struct wl_buffer *wlbuf);
 
 static GstIonBufFdMeta  *gst_wayland_sink_read_ionBufFd (GstWaylandSink *sink, GstBuffer * buf) {
@@ -797,11 +797,9 @@ struct wlbuffer_info
   unsigned offset;
 };
 
-gboolean check_event_struct (const GstStructure *s, GstIonBufFdMeta * bufinfo)
+gboolean check_event_struct (const GstStructure *s, gpointer bufinfo)
 {
-  /* gpointer * ionbufp = g_malloc(sizeof *ionbufp); */
-  GstIonBufFdMeta * ionbufp = g_malloc(sizeof * ionbufp);
-  unsigned size, fd, offset;
+  gpointer ionbufp;
   const GValue *ionbufval;
 
   ionbufval = gst_structure_get_value(s, "ion_buf_pointer");
@@ -810,21 +808,14 @@ gboolean check_event_struct (const GstStructure *s, GstIonBufFdMeta * bufinfo)
       GST_ERROR("check_event_struct: No GValue in wayland's event structure.");
       return FALSE;
     }
-  ionbufp = (GstIonBufFdMeta *) g_value_get_pointer(ionbufval);
+  ionbufp = g_value_get_pointer(ionbufval);
   if (!ionbufp)  {
     GST_ERROR("No readable GstIonBufMeta data.");
     return FALSE;
   }
-  /*  gst_structure_get_uint(s, "ion_buf_size", &size);
-  gst_structure_get_uint(s, "ion_buf_fd", &fd);
-  gst_structure_get_uint(s, "ion_buf_offset", &offset);
-  g_print("\tGstOMXBuffer * ionbufp %p, ion_buf_size %u, ion_buf_fd %u, ion_buf_offset %u\n.", ionbufp, size, fd, offset);
-  g_assert(ionbufp == bufinfo->ionbufp);
-  g_assert(size == bufinfo->size);
-  g_assert(fd == bufinfo->fd);
-  g_assert(bufoffset == bufinfo->offset); */
 
-  GST_DEBUG("\tGstOMXBuffer * ionbufp %p.", ionbufp);
+  if (ionbufp != bufinfo)
+  GST_ERROR("buffer_release event consistency check\n:\tGstOMXBuffer* from event  %p doesn't match direct value %p", ionbufp, bufinfo);
 
   return TRUE;
 }
@@ -836,7 +827,7 @@ wl_omx_buffer_release (void *data, struct wl_buffer *wlbuf)
   GstEvent * upstream_release = g_malloc(sizeof *upstream_release);
   GstPad *sinkpad = GST_BASE_SINK_PAD(bufinfo->sink);
   GstStructure * event_struct = gst_structure_new_empty("IonBufRelease");
-  gboolean pushed;
+  gboolean ret;
 
   GValue bufval = { 0, };
   g_value_init(&bufval, G_TYPE_POINTER);
@@ -851,13 +842,17 @@ wl_omx_buffer_release (void *data, struct wl_buffer *wlbuf)
       return;
   }
 
-  check_event_struct((const GstStructure *) event_struct, (GstIonBufFdMeta *) bufinfo->msg_data);
+  ret = check_event_struct((const GstStructure *) event_struct, bufinfo->msg_data);
 
-  upstream_release = gst_event_new_custom (GST_EVENT_CUSTOM_UPSTREAM, event_struct);
-  pushed = gst_pad_push_event (sinkpad, upstream_release);
-  g_assert (pushed != FALSE);
-
-  g_value_unset(&bufval);
+  if (ret) {
+    upstream_release = gst_event_new_custom (GST_EVENT_CUSTOM_UPSTREAM, event_struct);
+    ret = gst_pad_push_event (sinkpad, upstream_release);
+  } else {
+    GST_ERROR("Failed to push wl_omx_buffer_release() event for %p", bufinfo->msg_data);
+    gst_event_unref(upstream_release);
+    gst_structure_free(event_struct);
+    g_value_unset(&bufval);
+  }
 
   return;
 }
-- 
1.9.1

