diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe0d955..6d51a57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,5 +62,5 @@ endif(SCTP_SUPPORT)
 set_target_properties(addsources PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(cannelloni addsources cannelloni-common pthread)
 
-install(TARGETS cannelloni DESTINATION bin)
-install(TARGETS cannelloni-common DESTINATION lib)
+install(TARGETS cannelloni DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS cannelloni-common DESTINATION ${CMAKE_INSTALL_LIBDIR})
