From 49eeee1d58b8f57a3a68df3968030f9fd8c4f1ac Mon Sep 17 00:00:00 2001
From: Venkat Raju sana <quic_vrsana@quicinc.com>
Date: Sat, 9 Jul 2022 16:21:45 -0700
Subject: [PATCH 1/2] Fix for build issue

Signed-off-by: Venkat Raju sana <quic_vrsana@quicinc.com>
---
 include/linux/psi.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/psi.h b/include/linux/psi.h
index 65eb1476ac70..926762afa76e 100644
--- a/include/linux/psi.h
+++ b/include/linux/psi.h
@@ -55,7 +55,7 @@ static inline void psi_cgroup_free(struct cgroup *cgrp)
 }
 static inline void cgroup_move_task(struct task_struct *p, struct css_set *to)
 {
-	rcu_assign_pointer(p->cgroups, to);
+//	rcu_assign_pointer(p->cgroups, to);
 }
 #endif
 
-- 
2.17.1

