From 46bc486f9ff03af8cb1326280b19aac581981b34 Mon Sep 17 00:00:00 2001
From: Yannick Gicquel <yannick.gicquel@iot.bzh>
Date: Wed, 28 Sep 2016 09:48:31 +0200
Subject: [PATCH 5/9] node: fixup default string for agl_player nodes.

Signed-off-by: Yannick Gicquel <yannick.gicquel@iot.bzh>
---
 node.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/node.c b/node.c
index c46f842..8ebfbc1 100644
--- a/node.c
+++ b/node.c
@@ -157,7 +157,7 @@ agl_node_type agl_node_type_from_str (const char *str)
 
 	if (pa_streq (str, "agl_radio"))
 		type = agl_radio;
-	else if (pa_streq (str, "agl_music"))
+	else if (pa_streq (str, "agl_player"))
 		type = agl_player;
 	else if (pa_streq (str, "agl_navigator"))
 		type = agl_navigator;
-- 
1.9.1

