From 44d287160070bca58dba581d9bcd9f6f403708ff Mon Sep 17 00:00:00 2001
From: Rahul Agarwal <quic_rahaga@quicinc.com>
Date: Wed, 16 Oct 2024 07:07:11 +0000
Subject: [PATCH] add diag user

---
 passwd.master | 1 +
 1 file changed, 1 insertion(+)

diff --git a/passwd.master b/passwd.master
index 66e9033..b19efe1 100644
--- a/passwd.master
+++ b/passwd.master
@@ -15,4 +15,5 @@ backup:*:34:34:backup:/var/backups:/usr/sbin/nologin
 list:*:38:38:Mailing List Manager:/var/list:/usr/sbin/nologin
 irc:*:39:39:ircd:/run/ircd:/usr/sbin/nologin
 _apt:*:42:65534::/nonexistent:/usr/sbin/nologin
+diag:*:53:53:diag:/nonexistent:/usr/sbin/nologin
 nobody:*:65534:65534:nobody:/nonexistent:/usr/sbin/nologin
-- 
2.44.1

