From d5c87961ab2a5845eaa02f91158a486a72d24afd Mon Sep 17 00:00:00 2001
From: Daniel Tak <ytak@codeaurora.org>
Date: Fri, 7 Dec 2018 12:57:36 -0700
Subject: [PATCH] add sdcard diag groups

---
 group.master | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/group.master b/group.master
index e54fd1d..0092a3a 100644
--- a/group.master
+++ b/group.master
@@ -40,4 +40,6 @@ games:*:60:
 shutdown:*:70:
 wheel:*:80:
 users:*:100:
+sdcard:*:1015:diag
+diag:*:53:
 nogroup:*:65534:
