From 09a0ef7e9fe40394850d3fedc59ed55c080a2d1e Mon Sep 17 00:00:00 2001
From: Lei wang <leiwan@codeaurora.org>
Date: Mon, 4 Sep 2017 15:35:31 +0800
Subject: [PATCH] systemd: udev rules add Tag to diag device

Signed-off-by: Lei wang <leiwan@codeaurora.org>
---
 rules/99-systemd.rules.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rules/99-systemd.rules.in b/rules/99-systemd.rules.in
index 5c2cda5..17e5fe8 100644
--- a/rules/99-systemd.rules.in
+++ b/rules/99-systemd.rules.in
@@ -64,4 +64,6 @@ SUBSYSTEM=="misc", KERNEL=="rfkill", TAG+="systemd", ENV{SYSTEMD_WANTS}+="system
 SUBSYSTEM=="module", KERNEL=="fuse", TAG+="systemd", ENV{SYSTEMD_WANTS}+="sys-fs-fuse-connections.mount"
 SUBSYSTEM=="module", KERNEL=="configfs", TAG+="systemd", ENV{SYSTEMD_WANTS}+="sys-kernel-config.mount"
 
+SUBSYSTEM=="diag", KERNEL=="diag", TAG+="systemd"
+
 LABEL="systemd_end"
-- 
1.9.1

