diff --git a/arch/arm/boot/dts/qcom/msm8996-pinctrl.dtsi b/arch/arm/boot/dts/qcom/msm8996-pinctrl.dtsi
index 50312a5..26440ce 100644
--- a/arch/arm/boot/dts/qcom/msm8996-pinctrl.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996-pinctrl.dtsi
@@ -192,6 +192,22 @@
 			};
 		};
 
+		pmx_vidc: pmx_vidc {
+			vidc_0_switch_active: vidc_0_switch_active {
+				mux {
+					pins = "gpio109";
+					function = "gpio";
+					bias-pull-down; /* pull down */
+				};
+
+				config {
+					pins = "gpio109";
+					drive-strength = <4>; /* 16 mA */
+					bias-pull-down; /* pull down */
+				};
+			};
+		};
+
 		uart_console_active: uart_console_active {
 			mux {
 				pins = "gpio4", "gpio5";
diff --git a/arch/arm/boot/dts/qcom/msm8996-vidc.dtsi b/arch/arm/boot/dts/qcom/msm8996-vidc.dtsi
old mode 100644
new mode 100755
index 21aa1db4..8d35871
--- a/arch/arm/boot/dts/qcom/msm8996-vidc.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8996-vidc.dtsi
@@ -17,6 +17,8 @@
 		compatible = "qcom,msm-vidc";
 		status = "ok";
 		reg = <0xC00000 0xff000>;
+		/* uim1 ldo */
+		vdd-supply = <&pm8994_l9>;
 		interrupts = <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>;
 		qcom,hfi = "venus";
 		qcom,hfi-version = "3xx";
@@ -94,7 +96,10 @@
 		       <&clock_mmss clk_video_subcore0_clk>,
 		       <&clock_mmss clk_video_subcore1_clk>;
 		qcom,clock-configs = <0x0 0x0 0x0 0x1 0x0 0x0 0x0 0x1 0x1>;
+    pinctrl-names = "pmx_vidc_active";
+		pinctrl-0 = <&vidc_0_switch_active>;
 
+		adi,switch-gpio = <&tlmm 109 0x1>;
 		/* Buses */
 		bus_cnoc {
 			compatible = "qcom,msm-vidc,bus";
diff --git a/drivers/media/platform/msm/vidc/msm_v4l2_vidc.c b/drivers/media/platform/msm/vidc/msm_v4l2_vidc.c
old mode 100644
new mode 100755
index e81162f..e67c850
--- a/drivers/media/platform/msm/vidc/msm_v4l2_vidc.c
+++ b/drivers/media/platform/msm/vidc/msm_v4l2_vidc.c
@@ -24,6 +24,10 @@
 #include <linux/version.h>
 #include <linux/io.h>
 #include <media/msm_vidc.h>
+#include <linux/regulator/consumer.h>
+#include <linux/regulator/driver.h>
+#include <linux/gpio.h>
+#include <linux/of_gpio.h>
 #include "msm_vidc_common.h"
 #include "msm_vidc_debug.h"
 #include "msm_vidc_internal.h"
@@ -446,7 +450,77 @@ static const struct of_device_id msm_vidc_dt_match[] = {
 	{.compatible = "qcom,msm-vidc,bus"},
 	{}
 };
+static int  vidc_parse_dt(struct device *dev)
+{
+	struct device_node *np = dev->of_node;
+	struct device_node *reg_node = NULL;
+	struct regulator *reg = NULL;
+	int ret = 0;
+
+	struct pinctrl *ts_pinctrl = NULL;
+	struct pinctrl_state *pinctrl_state_active = NULL;
+
+	u32 switch_gpio;
+	u32 switch_flags;
+
+	reg_node = of_parse_phandle(np, "vdd-supply", 0);
+	if (reg_node) {
+		reg = devm_regulator_get(dev, "vdd");
+		if (IS_ERR(reg)) {
+			printk("%s: no vdd!\n", __func__);
+		} else {
+			printk("%s: found vdd!\n", __func__);
+			regulator_set_voltage(reg, 1800000, 1800000);
+			ret = regulator_enable(reg);
+			if (ret)
+				printk("fail to enable ldo for rvc trigger gpio\n");
+		}
+		of_node_put(reg_node);
+	}
 
+	/* Get pinctrl if target uses pinctrl */
+	ts_pinctrl = devm_pinctrl_get(dev);
+	if (IS_ERR_OR_NULL(ts_pinctrl)) {
+		ret = PTR_ERR(ts_pinctrl);
+		printk("%s: Pincontrol DT property returned %X\n",
+			__func__, ret);
+	}
+
+	pinctrl_state_active = pinctrl_lookup_state(ts_pinctrl,
+		"pmx_vidc_active");
+	if (IS_ERR_OR_NULL(pinctrl_state_active)) {
+		ret = PTR_ERR(pinctrl_state_active);
+		printk("Can not lookup pmx_vidc_active pinstate %d\n", ret);
+	}
+
+	ret = pinctrl_select_state(ts_pinctrl, pinctrl_state_active);
+	if (ret < 0)
+		printk("%s: Failed to select pmx_vidc_active pinstate %d\n",
+			__func__, ret);
+
+	switch_gpio = of_get_named_gpio_flags(np,
+				"adi,switch-gpio", 0, &switch_flags);
+	if (gpio_is_valid(switch_gpio)) {
+		ret = gpio_request(switch_gpio,
+			"vidc_switch_gpio");
+		if (ret) {
+			printk("%d unable to request gpio [%d] ret=%d\n",
+					__LINE__, switch_gpio, ret);
+			goto exit;
+		}
+
+		ret = gpio_direction_input(switch_gpio);
+		if (ret) {
+			printk("unable to set dir for gpio [%d]\n",
+					switch_gpio);
+		}
+
+		gpio_free(switch_gpio);
+	}
+
+exit:
+	return ret;
+}
 static int msm_vidc_probe_vidc_device(struct platform_device *pdev)
 {
 	int rc = 0;
@@ -607,7 +681,7 @@ static int msm_vidc_probe_vidc_device(struct platform_device *pdev)
 		dprintk(VIDC_ERR, "Failed to trigger probe for sub-devices\n");
 		goto err_fail_sub_device_probe;
 	}
-
+    vidc_parse_dt(&pdev->dev);
 	return rc;
 
 err_fail_sub_device_probe:
