From e78391b5b43ef24420b3403ae3e771651383863a Mon Sep 17 00:00:00 2001
From: Viswanath Kraleti <vkraleti@codeaurora.org>
Date: Sat, 21 Oct 2017 10:04:58 +0530
Subject: [PATCH] Change WORKSPACE references relative to srcdir

---
 Makefile.am | 23 ++++++++++++-----------
 1 file changed, 12 insertions(+), 11 deletions(-)

diff --git a/gatt/Makefile.am b/gatt/Makefile.am
index 4b788bd..0a55d6f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,10 +1,12 @@
+oss_bt_path := $(top_srcdir)/../../../vendor/qcom/opensource/bluetooth
+
 AM_CPPFLAGS = -O2 \
               -D_GNU_SOURCE \
-             -I${WORKSPACE}/qcom-opensource/bt/gatt/include \
-             -I${WORKSPACE}/qcom-opensource/bt/bt-app/include \
-             -I${WORKSPACE}/qcom-opensource/bt/bt-app/ \
-             -I${WORKSPACE}/vendor/qcom/opensource/bluetooth/hal/include/ \
-             -I${WORKSPACE}/vendor/qcom/opensource/bluetooth/vhal/include \
+             -I$(srcdir)/include \
+             -I$(srcdir)/../bt-app \
+             -I$(srcdir)/../bt-app/include \
+             -I$(oss_bt_path)/hal/include \
+             -I$(oss_bt_path)/vhal/include \
              -I${WORKSPACE}/system/bt \
              -I${WORKSPACE}/system/core/include

@@ -14,12 +16,11 @@ AM_CFLAGS = -Wall \
         -Wno-trigraphs \
         -DLINUX \
         -std=c99 \
-        -I${WORKSPACE}/qcom-opensource/bt/gatt/include \
-        -I${WORKSPACE}/qcom-opensource/bt/bt-app/include \
-        -I${WORKSPACE}/qcom-opensource/bt/bt-app/ \
-        -I${WORKSPACE}/vendor/qcom/opensource/bluetooth/hal/include/ \
-        -I${WORKSPACE}/vendor/qcom/opensource/bluetooth/vhal/include \
-        -I${WORKSPACE}/system/core/include
+        -I$(srcdir)/include \
+        -I$(srcdir)/../bt-app \
+        -I$(srcdir)/../bt-app/include \
+        -I$(oss_bt_path)/hal/include \
+        -I$(oss_bt_path)/vhal/include

 ACLOCAL_AMFLAGS = -I m4

--
1.9.1

