From d82c6727cdfce1dc82d9e5a3a7bd910ef219d433 Mon Sep 17 00:00:00 2001
From: Shabbir Limdiwala <quic_slimdiwa@qti.qualcomm.com>
Date: Thu, 21 Sep 2023 17:24:06 -0700
Subject: [PATCH] Restore pre generated hex files

Signed-off-by: Shabbir Limdiwala <quic_slimdiwa@qti.qualcomm.com>
---
 hw/i386/Makefile.objs | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/hw/i386/Makefile.objs b/hw/i386/Makefile.objs
index bd4f147f..e0a82f4a 100644
--- a/hw/i386/Makefile.objs
+++ b/hw/i386/Makefile.objs
@@ -13,19 +13,9 @@ hw/i386/acpi-build.o: hw/i386/acpi-build.c \
 iasl-option=$(shell if test -z "`$(1) $(2) 2>&1 > /dev/null`" \
     ; then echo "$(2)"; else echo "$(3)"; fi ;)
 
-ifdef IASL
-#IASL Present. Generate hex files from .dsl
-hw/i386/%.hex: $(SRC_PATH)/hw/i386/%.dsl $(SRC_PATH)/scripts/acpi_extract_preprocess.py $(SRC_PATH)/scripts/acpi_extract.py
-	$(call quiet-command, $(CPP) -x c -P $(QEMU_DGFLAGS) $(QEMU_INCLUDES) $< -o $*.dsl.i.orig, "  CPP $(TARGET_DIR)$*.dsl.i.orig")
-	$(call quiet-command, $(PYTHON) $(SRC_PATH)/scripts/acpi_extract_preprocess.py $*.dsl.i.orig > $*.dsl.i, "  ACPI_PREPROCESS $(TARGET_DIR)$*.dsl.i")
-	$(call quiet-command, $(IASL) $(call iasl-option,$(IASL),-Pn,) -vs -l -tc -p $* $*.dsl.i $(if $(V), , > /dev/null) 2>&1 ,"  IASL $(TARGET_DIR)$*.dsl.i")
-	$(call quiet-command, $(PYTHON) $(SRC_PATH)/scripts/acpi_extract.py $*.lst > $*.off, "  ACPI_EXTRACT $(TARGET_DIR)$*.off")
-	$(call quiet-command, cat $*.off > $@, "  CAT $(TARGET_DIR)$@")
-else
-#IASL Not present. Restore pre-generated hex files.
+#Restore pre-generated hex files.
 hw/i386/%.hex: $(SRC_PATH)/hw/i386/%.hex.generated
 	$(call quiet-command, cp -f $< $@, "  CP $(TARGET_DIR)$@")
-endif
 
 .PHONY: cleanhex
 cleanhex:
-- 
2.17.1

