Update texi2pod.pl for perl 5.26 compatibility

From: Sidharth Mehta <quic_sidhmeht@quicinc.com>

--- a/scripts/texi2pod.pl	2015-08-11 12:11:10.000000000 -0700
+++ b/scripts/texi2pod.pl	2023-07-18 15:26:44.712204450 -0700
@@ -317,7 +317,7 @@
 	@columns = ();
 	for $column (split (/\s*\@tab\s*/, $1)) {
 	    # @strong{...} is used a @headitem work-alike
-	    $column =~ s/^\@strong{(.*)}$/$1/;
+	    $column =~ s/^\@strong\{(.*)\}$/$1/;
 	    push @columns, $column;
 	}
 	$_ = "\n=item ".join (" : ", @columns)."\n";
